vm_automation/esxiVm.py

```diff
@@ -220,6 +220,42 @@ def __init__(self, serverObject, vmObject):
         else:
             self.arch = 'x86'
 
+    def waitForVmToBoot(self):
+        # IS IT TURNED ON?
+        retVal = True
+        for i in range(10):
+            if self.isPoweredOn() == False:
+                time.sleep(1)
+                self.server.logMsg(self.vmName + " WAS NOT POWERED ON AS EXPECTED; RETRYING")
+                self.powerOn(True)
+                sleep(5)
+        # IS THE TOOLS SERVICE RUNNING?
+        if self.isPoweredOn() == False:
+            return False
+        else:
+            attempts = 300
+            for i in range(attempts):

```

Comment:
This seems to just bash at the server for 300 requests, should it be changed to simply add a delay when not ready like all the other loops here?

```
if self.checkTools(True) != 'TOOLS_NOT_READY':
    time.sleep(5)
    break
```

Or maybe for this whole loop

```
tools_ready = False
for i in range(attempts)
    if self.checkTools(True) == 'TOOLS_NOT_READY':
        tools_ready = True
        break
    time.sleep(5)
# No tools available return as result.
if !tools_ready
    return tools_ready
```

Slightly more efficient with one less integer subtraction and test per loop and adds the `sleep`